provider/string_util.go

```diff
@@ -0,0 +1,16 @@
+package provider
+
+import "strings"
+
+func splitAndTrimString(separatedString string) []string {
```

Comment:
I just realize we don't have any dedicated tests for this function. Could you please add some? That way, we guarantee the function does what it should independent of any of its current users.